dvc/machine/__init__.py

```diff
@@ -200,3 +203,17 @@ def get_sshfs(self, name: Optional[str]):
     def run_shell(self, name: Optional[str]):
         config, backend = self.get_config_and_backend(name)
         return backend.run_shell(**config)
+
+    def rename(self, old_name: str, new_name: str):
+        """move configuration to a new location if the machine is running."""
+        from dvc.config import MachineExistError
+
+        cloud_type = self._get_config(name=old_name)["cloud"]
+        backend = self.CLOUD_BACKENDS[cloud_type]
+        old_path = os.path.join(self.machine_space, backend, old_name)
+        new_path = os.path.join(self.machine_space, backend, new_name)
+        if os.path.exists(new_path):
+            raise MachineExistError(new_name, new_path)
+        if not os.path.exists(old_path):
+            return
+        move(old_path, new_path)
```

Comment:
I think this should be handled in the `MachineBackend` classes instead of at the manager level. Moving the directory may not necessarily be the intended operation here, and the specifics of the `machine/<backend>` directory structure is intended to only have meaning for a given backend. 

Also, the backend will likely have to do some additional operations beyond just moving/renaming the directory. (Such as accounting for already-running machine instances that use the old name)